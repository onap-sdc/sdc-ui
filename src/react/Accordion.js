import React from 'react';
import PropTypes from 'prop-types';
import SVGIcon from './SVGIcon.js';

class Accordion extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: false
    };
  }
  render() {
    const { children, title, className } = this.props;
    const { open } = this.state;
    return (
      <div className={`sdc-accordion ${className}`}>
        <div onClick={() => this.setState({ open: !open })} className='sdc-accordion-header'>
          <SVGIcon name='chevronUp' iconClassName={open ? 'down' : ''} />
          <div className='title'>{title}</div>
        </div>
        <div className={`sdc-accordion-body ${open ? 'open' : ''}`}>{children}</div>
      </div>
    );
  }
}

Accordion.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node
};

Accordion.defaultProps = {
  title: '',
  className: ''
};

export default Accordion;
