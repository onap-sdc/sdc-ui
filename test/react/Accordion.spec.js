import React from 'react';
import Accordion from '../../src/react/Accordion.js';

import renderer from 'react-test-renderer';

describe('Accordion', () => {
	test('Accordion - Default', () => {
		const accordion = renderer.create(<Accordion title='Accordion'>Accordion body</Accordion>).toJSON();
		expect(accordion).toMatchSnapshot();
	});
});