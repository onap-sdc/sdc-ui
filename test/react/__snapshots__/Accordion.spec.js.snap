// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion Accordion - Default 1`] = `
<div
  className="sdc-accordion "
>
  <div
    className="sdc-accordion-header"
    onClick={[Function]}
  >
    <div
      className="svg-icon-wrapper     bottom"
      disabled={undefined}
      onClick={undefined}
    >
      <svg />
      
    </div>
    <div
      className="title"
    >
      Accordion
    </div>
  </div>
  <div
    className="sdc-accordion-body "
  >
    Accordion body
  </div>
</div>
`;
